### PR TITLE
Gives the detective a new gun, changes some sprites, and refluffs the 9mm pistol.

### PR DESCRIPTION
**Pull request by TheDanseMacabre**
https://www.youtube.com/watch?v=GyZnEVUDREM&feature=youtu.be

The Detective's revolver was something of a pariah - A relic item from the noir days. It used snowflake ammunition that could not be acquired from anywhere but cargo and the detective's office. It had no fluff, other than the fact it was apparently based off of the Smith and Wesson Model 10, a 550 year old revolver now woefully obsolete. I have opted to replace it with a rather nice .45, which fills the same niche as the revolver did but with more standardization with other security equipment. Also gives the detective less ammunition overall, 14 bullets instead of 18. This is balanced out by the fact that the .45 does slightly more damage.

Credits to Amory for the sprites.

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 1]
```diff
@@ -80,6 +80,22 @@
 	else
 		icon_state = "secgundark-e"
 
+/obj/item/gun/projectile/sec/detective
+	desc = "A compact NanoTrasen designed sidearm, popular with law enforcement personnel for concealed carry purposes. It has a faux wooden grip. Uses .45 rounds."
+	description_fluff = "The NT Mk10 Inspector is a ballistic sidearm developed and produced by Nanotrasen. Unlike the related Mk58, the Inspector is a rather high quality piece - typically issued to higher ranking law enforcement personnel, the Mk10 is compact and chambered in .45 caliber. With all the bells and whistles of a modern, quality police pistol, the Inspector's main drawback is the notoriously nippy recoil - .45 in such a small package does not play well with the average shooter."
+	name = "compact .45 pistol"
+	icon_state = "detgun"
+	item_state = "detgunheld"
+	w_class = 2
```
Comment (mikomyazaki) [inline code/modules/projectiles/guns/projectile/pistol.dm]:
```suggestion
	w_class = ITEMSIZE_SMALL
```

File: code/modules/projectiles/guns/projectile/pistol.dm [hunk 2]
```diff
@@ -80,6 +80,22 @@
...
+	magazine_type = /obj/item/ammo_magazine/c45m
+
+/obj/item/gun/projectile/sec/detective/update_icon()
+	..()
+	if(ammo_magazine && ammo_magazine.stored_ammo.len)
```
Comment (mikomyazaki) [inline code/modules/projectiles/guns/projectile/pistol.dm]:
```suggestion
	if(ammo_magazine?.stored_ammo.len)
```

**Comment by TheDanseMacabre** [comment]
!review

**Comment by alsoandanswer** [comment]
was there a feedback thread

**Comment by TheDanseMacabre** [comment]
@alsoandanswer is that necessary for a change as minute as this

**Comment by alsoandanswer** [comment]
it is up to the community whether the change is "minute" or not.

**Pull request closed without merging**